From c08f67730aba342b03f070209acc2990d3decf3c Mon Sep 17 00:00:00 2001
From: Ming Lei <ming.lei@canonical.com>
Date: Fri, 17 Aug 2012 22:06:58 +0800
Subject: [PATCH] firmware loader: fix compile failure if !PM
Git-commit: c08f67730aba342b03f070209acc2990d3decf3c
Patch-mainline: v3.7-rc1
Reference: fate#314574
Target: sle11-sp3

'return 0' should be added to fw_pm_notify if !PM because
return value of the funcion is defined as 'int'.

Reported-by: Fengguang Wu <fengguang.wu@intel.com>
Signed-off-by: Ming Lei <ming.lei@canonical.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Acked-by: Lee, Chun-Yi <jlee@suse.com>

---
 drivers/base/firmware_class.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/base/firmware_class.c
+++ b/drivers/base/firmware_class.c
@@ -1244,7 +1244,9 @@ static int fw_pm_notify(struct notifier_
 #else
 static int fw_pm_notify(struct notifier_block *notify_block,
 			unsigned long mode, void *unused)
-{}
+{
+	return 0;
+}
 #endif
 
 static void __init fw_cache_init(void)
